### PR TITLE
added User entity and related classes & interfaces:

### PR DESCRIPTION
**Pull request by DanilKi**
- added DTOs for handling http requests & responses;
 - added repository, mapper, service for User entity;
 - added AuthController with registration endpoint;
 - added RegistrationException and its handling method to GlobalExceptionHandler;
 - added FieldMatch annotation for validating matches of two fields of an object.

### CODE REVIEW

File: src/main/java/com/onlinebookstore/exception/CustomGlobalExceptionHandler.java [hunk 1]
```diff
@@ -53,6 +53,18 @@ protected ResponseEntity<Object> handleEntityNotFound(RuntimeException ex,
...
+    @ResponseStatus(HttpStatus.NOT_ACCEPTABLE)
+    @ExceptionHandler(value = RegistrationException.class)
+    protected ResponseEntity<Object> handleRegistrationException(Exception ex,
+                                                                 WebRequest request) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put(TIME_STAMP, LocalDateTime.now());
+        body.put(HTTP_STATUS, HttpStatus.NOT_ACCEPTABLE);
+        body.put(ERRORS_MSG, ex.getLocalizedMessage());
+        return handleExceptionInternal(ex, body, new HttpHeaders(),
+                HttpStatus.NOT_ACCEPTABLE, request);
```
Comment (Elena-Bruyako) [inline src/main/java/com/onlinebookstore/exception/CustomGlobalExceptionHandler.java]:
```suggestion
                HttpStatus.CONFLICT, request);
```

File: src/main/java/com/onlinebookstore/service/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto)
+            throws RegistrationException {
+        if (userRepository.findByEmail(requestDto.getEmail()).isPresent()) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/onlinebookstore/service/impl/UserServiceImpl.java]: It's better to use `boolean existsByEmail(String email)`

File: src/main/resources/db/changelog/changes/02-create-users-table.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            columns:
+              - column:
+                  name: id
+                  type: bigint
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: email
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/02-create-users-table.yml:17]: Be sure, that the size in the same as in DTO fields

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's fix some mistakes

**Review by MentorMateAcademy — APPROVED** [review]

**Pull request merged**